### DISCUSSION TITLE
Filtering continues after stop (not the bug)

### DISCUSSION THREAD

**Discussion by TonyGravagno** [RealRaven2000/FiltaQuilla#290] [Q&A]
This is not about the issue with filters failing to process actions.

In the screenshot below we see FQ processing rules, taking an action, and then another filter is processed for its rules without the action taken. The action on the first filter definitely ends with Stop Filter Execution.

I'm not concerned that it's processing, since it's not doing anything else. I'm just thinking about performance when rules continue to be processed when we know nothing more will be done.

Yes, i'm sure that's another filter that follows the first, and yes I'm sure it's the same mail item.

This is not an issue with FQ. I'm asking if this expected Tb behaviour ... which wouldn't have been seen except for the helpful logging by FQ! 😁 

![image](https://github.com/user-attachments/assets/d9813a94-05a7-4874-accf-69fea0bfbf74)

**Comment by RealRaven2000** [comment]
> This is not an issue with FQ. I'm asking if this expected Tb behaviour ... which wouldn't have been seen except for the helpful logging by FQ! 😁

well, who knows? I think Thunderbird staff may even have a hard time finding somebody who knows or remembers the answer to this question. I know that it's well above my paygrade.

**Comment by TonyGravagno** [comment]
I'm wondering if this is not synchronous reporting. I don't have a solid definition of the concept yet, but I see many indications that the logging may hit the console outside of the order in which it's triggered.

What do you think about adding an option to add this info to console notifications? :
 
- Message ID
- Timestamp (ms)
- Name of profile+filter being processed

This info would help us to know how filters are being processed, and if something is out of order an enhancement request can be submitted to Tb for log synchronization.
